belgif-rest-problem/src/test/java/io/github/belgif/rest/problem/internal/AnnotationUtilTest.java

```diff
@@ -0,0 +1,127 @@
+package io.github.belgif.rest.problem.internal;
+
+import static org.assertj.core.api.Assertions.*;
+
+import java.lang.annotation.Annotation;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.Optional;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+class AnnotationUtilTest {
+
+    @Test
+    void notFound() {
+        assertThat(AnnotationUtil.findParamAnnotation(getMethod("notFound"), 0, Foo.class)).isEmpty();
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = { "ownMethod", "interfaceMethod", "parentInterfaceMethod", "parentMethod",
+            "grandParentMethod", "overriddenParentMethod", "overriddenGrandParentMethod" })
+    void found(String method) {
+        Optional<? extends Annotation> annotation = AnnotationUtil.findParamAnnotation(getMethod(method), 0, Foo.class);
+        assertThat(annotation).isPresent();
+        assertThat(annotation.get()).isInstanceOf(Foo.class);
+        Foo foo = (Foo) annotation.get();
+        assertThat(foo.value()).isEqualTo(method);
+    }
+
+    @Test
+    void byName() {
+        Optional<? extends Annotation> annotation =
+                AnnotationUtil.findParamAnnotation(getMethod("ownMethod"), "i", Foo.class);
+        assertThat(annotation).isPresent();
+    }
+
+    @Test
+    void byNameFallback() {
+        Optional<? extends Annotation> annotation =
+                AnnotationUtil.findParamAnnotation(getMethod("ownMethod"), "fallback", Foo.class);
```

Comment:
This fallback behavior isn't described in the javadoc, reading it I was under the impression that an empty `Optional` would be returned but that isn't the case.

Reply:
This fallback was needed to get the nestedQueryParamsViolation and doubleNestedQueryParamsViolation tests working.
In those tests, it's not really the method parameter name that gets passed, but the field name
(from io.github.belgif.rest.problem.spring.BeanValidationExceptionsHandler#handleBindException). Maybe it can be handled differently.

For now, describing this fallback in the javadoc.